hedera-node/docs/design/services/smart-contract-service/update_token_custom_fees_sc.md

```diff
@@ -0,0 +1,87 @@
+# Update Token Custom Fees via Smart Contracts
+
+## Purpose
+
+In order to support updating the Token Custom Fees via smart contract calls, we need to expose new smart contract functions in the Hedera Smart Contract Service (HSCS) logic exposing Hedera Token Service (HTS) logic. This document will define the architecture and implementation of the `updateFungibleTokenCustomFees` and `updateNonFungibleTokenCustomFees` smart contract functions that will extend tha capabilities of the HSCS.
+
+## References
+
+[HIP-18](https://hips.hedera.com/hip/hip-18) - HIP that introduces token custom fees.\
+[HIP-206](https://hips.hedera.com/hip/hip-206) - HIP that enables the precompiled system contracts for Hedera Token Service.\
+[HIP-514](https://hips.hedera.com/hip/hip-514) - HIP that introduces token management via smart contracts.
+
+## Goals
+
+- Expose `updateFungibleTokenCustomFees` and `updateNonFungibleTokenCustomFees` as new functions in the Hedera Token Service Smart Contract.
+- Implement the needed HTS system contract classes to support the new function.
+
+## Non Goals
+
+- The implementation of the HAPI operation, as it is already an existing feature.
+
+## Architecture
+
+The architecture for update token custom fees follows the existing framework defined for handling all calls to the HederaTokenService system contract in the modularization services and is described in more detail in the Implementation section below.
+
+## Implementation
+
+### New Function Implementations
+
+New system contract functions must be added to the `IHederaTokenService` interface to support the updating the custom fees for fungible and non-fungible tokens.
+
+| Function Selector Hash |                                                                                        Function Signature                                                                                         |    Response    |                                 |
+|------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------|---------------------------------|
+| `0xe780c5d3`           | `function updateFungibleTokenCustomFees(address token,  IHederaTokenService.FixedFee[] memory fixedFees, IHederaTokenService.FractionalFee[] memory fractionalFees) returns (int64 responseCode)` | `ResponseCode` | The response code from the call |
```

Comment:
I know our other design docs don't do this (nor do the HIPs) but given the way the selector is computed it might be nice to also list the fully expanded signature (I mean, the one where all the types like `FixedFee` are expanded out so all that's left is a native Solidity signature).

Reply:
It is already merged like this with the HIP, but we can start doing it like this from now on.

Reply:
This is a design doc, you can have extra information here that's not in the HIP (as long as it is consistent, of course!). But up to you.